spec/system/discussion_spec.rb

```diff
@@ -0,0 +1,89 @@
+require "rails_helper"
+require_relative "../modules/helpers"
+require_relative "../matchers/category_matchers"
+require_relative "../matchers/discussion_matchers"
+require_relative "../matchers/shared_matchers"
+
+RSpec.describe "Discussion,", type: :system do
+  include SharedMatchers
+  include SessionHelpers
+  include CategoryHelpers
+  include DiscussionHelpers
+
+  context "when logged in as admin," do
+    before(:each) do
+      login_as("admin")
+    end
+
+    let(:category_name) { "Test Category Name" }
+    let(:discussion_title) { "Test Discussion Title" }
+    let(:discussion_body) { "Test discussion body" }
+
+    it "can be created" do
+      create_discussion(discussion_title, discussion_body, Category.create(name: category_name, position: 1, description: CATEGORY_DESCRIPTION))
+      discussion = Discussion.find_by(title: discussion_title, body: discussion_body)
+
+      expect(page).to have_link_tree(category_name, discussion_title)
+      expect(page).to have_correct_discussion_title_on_page(discussion_title)
+      expect(page).to have_correct_discussion_message_for_action("created")
+      expect(page).to have_expected_discussion_buttons
+      within "#comment-section" do
+        expect(page).to have_write_comment_section
+      end
+      within "#user-info" do
+        expect(page).to have_correct_user_info(discussion)
+      end
+      within "#link-tree" do
+        click_link category_name
+      end
+      expect(page).to have_the_expected_category_header("DISCUSSION REPLIES LAST POST")
+      within "#discussion-#{discussion.id}" do
+        find_by_id("discussion-#{discussion.id}-link")
+      end
+      within "#discussion-#{discussion.id}-info" do
+        expect(page).to have_text(discussion.user.login)
+        expect(page).to have_text(discussion.created_at.strftime("%m/%d/%Y %I:%M%p"))
+      end
+      within "#discussion-#{discussion.id}-replies .reply-counter" do
+        expect(page).to have_text(discussion.comments.count)
+      end
+    end
+
+    it "can be edited" do
+      updated_title = "New Title"
+      updated_body = "New Body"
+
+      # create_discussion(discussion_title, discussion_body)
+      category = Category.create(name: category_name, position: 1, description: CATEGORY_DESCRIPTION)
+      user = User.last
```

Comment:
Perhaps move this to a `let` call?  that'll remove the need for this line and in the deletion test.